packages/ui/Scrollable/Scrollable.js

```diff
@@ -693,79 +699,74 @@ class ScrollableBase extends Component {
 		return clamp(0, 1, 2 * overDistance / baseSize);
 	}
 
-	updateOverscrollEffectOnPosition = (targetX, targetY, type, ratio) => {
-		this.updateOverscrollEffect('horizontal', targetX, type, ratio);
-		this.updateOverscrollEffect('vertical', targetY, type, ratio);
+	// This should be called by user interaction
+	updateOverscrollEffectOnPosition = (targetX, targetY, type) => {
+		this.updateOverscrollEffect('horizontal', targetX, type);
+		this.updateOverscrollEffect('vertical', targetY, type);
 	}
 
-	updateOverscrollEffectByDirection = (orientation, position, type, ratio = null) => {
+	// This can be called by user interaction or on-going scrolling
+	updateOverscrollEffectByDirection = (orientation, position, type, ratio) => {
 		const applyOverscrollEffect = this.props.applyOverscrollEffect;
 
 		if (applyOverscrollEffect && this.getScrollabilities(orientation)) {
 			const
 				isVertical = (orientation === 'vertical'),
 				curPos = isVertical ? this.scrollTop : this.scrollLeft,
-				maxPos = this.getScrollBounds()[isVertical ? 'maxTop' : 'maxLeft'];
-			let
-				newType = type,
-				newRatio = (ratio === null) ? 1 : ratio;
-
-			if (newRatio > 0) {
-				if (type === overscrollTypes.scrolling) {
-					if (position === 'before' && curPos <= 0) { // Already on the beginning egde
-						applyOverscrollEffect(orientation, 'before', type, ratio);
-						newType = overscrollTypes.none;
-						newRatio = 0;
-					} else if (position >= 'after' && curPos >= maxPos) { // Already on the ending egde
-						applyOverscrollEffect(orientation, 'after', type, ratio);
-						newType = overscrollTypes.none;
-						newRatio = 0;
-					}
-				} else if (type === overscrollTypes.sticking) {
-					if (position === 'before' && curPos <= 0) { // Already on the beginning egde
-						applyOverscrollEffect(orientation, 'before', type, ratio);
-					} else if (position === 'after' && curPos >= maxPos) { // Already on the ending egde
-						applyOverscrollEffect(orientation, 'after', type, ratio);
-					}
+				maxPos = this.getScrollBounds()[isVertical ? 'maxTop' : 'maxLeft'],
+				{none, scrolling} = overscrollTypes;
+
+			if (ratio > 0 && type === scrolling) {
+				if (position === 'before' && curPos <= 0) { // Already on the beginning egde
+					applyOverscrollEffect(orientation, 'before', type, ratio);
+					this.setOverscrollStatus(orientation, position, none, 0);
```

Comment:
Why don't you use `'before'` string instead of position variable for code readability?

Reply:
Okay.